src/Cake.NuGet/NuGetPackageInstaller.cs

```diff
@@ -175,6 +160,38 @@ public IReadOnlyCollection<IFile> Install(PackageReference package, PackageType
             return result;
         }
 
+        private static DirectoryPath GetPackagePath(DirectoryPath root, PackageReference package)
+        {
+            if (package.Parameters.ContainsKey("version"))
+            {
+                var version = package.Parameters["version"].First();
+                return root.Combine($"{package.Package}.{version}");
```

Comment:
Well seems to be package dependent as GitReleaseManager 
```xml
<version>0.6.0</version>
``` 

`nuget install gitreleasemanager -Version 0.6.0`
![image](https://user-images.githubusercontent.com/1647294/28784223-0dccd268-7613-11e7-9a0c-8149a3ea2f6f.png)